tensorflow_addons/activations/tests/rrelu_test.py

```diff
@@ -23,43 +23,37 @@
 SEED = 111111
 
 
+def rrelu_wrapper(lower, upper, training):
+    def inner(x):
+        gs = tf.random.Generator.from_seed(SEED)
+        return rrelu(x, lower, upper, training=training, seed=None, gs=gs)
+
+    return inner
+
+
 @pytest.mark.parametrize("dtype", [np.float16, np.float32, np.float64])
 @pytest.mark.parametrize("training", [True, False])
 def test_rrelu(dtype, training):
     x = tf.constant([-2.0, -1.0, 0.0, 1.0, 2.0], dtype=dtype)
     lower = 0.1
     upper = 0.2
-
-    tf.random.set_seed(SEED)
-    result = rrelu(x, lower, upper, training=training, seed=SEED)
+    training_results = {
+        np.float16: [-0.3826, -0.165, 0, 1, 2],
+        np.float32: [-0.282151192, -0.199812651, 0, 1, 2],
+        np.float64: [-0.25720977, -0.1221586, 0, 1, 2],
+    }
+    gs = tf.random.Generator.from_seed(SEED)
+    result = rrelu(x, lower, upper, training=training, seed=None, gs=gs)
```

Comment:
We shouldn't modify an existing test when testing a new option. It should be a new standalone test. 

In the new test, rather than hardcoding the results, we could check that the generator works as expected, meaning that it doesn't affect the global random state, but is still reproductible. An example would be (pseudo code):

```python
def test1():
    """gs should ensure reproducibility without affecting the global random state."""
    tf.set_seed(0)
    a = tf.random.uniform()
    gs = tf.random.Generator.from_seed(8448)
    rrelu_result = rrelu(..., gs=gs)

    tf.set_seed(0)
    gs = tf.random.Generator.from_seed(8448)
    assert rrelu(..., gs=gs) == rrelu_result
    assert a == tf.random.uniform()
```

What do you think?

Reply:
I revert old test cases.